frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.stories.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { Meta, StoryObj, moduleMetadata } from '@storybook/angular';
+import { base64ModelDiagram } from 'src/storybook/encodedAssets';
+import { mockModel } from 'src/storybook/model';
+import { mockProject } from 'src/storybook/project';
+import { DiagramCacheMetadata } from '../service/model-diagram.service';
+import {
+  Diagrams,
+  ModelDiagramDialogComponent,
+} from './model-diagram-dialog.component';
+
+const meta: Meta<ModelDiagramDialogComponent> = {
+  title: 'Model Components / Model Diagram Cache',
+  component: ModelDiagramDialogComponent,
+};
+
+export default meta;
+type Story = StoryObj<ModelDiagramDialogComponent>;
+
+const emptyDiagramCacheMetadata: DiagramCacheMetadata = {
+  diagrams: [],
+  last_updated: '2024-04-29T14:00:00Z',
+};
+
+const loadedDiagramCacheMetadata: DiagramCacheMetadata = {
+  diagrams: [{ name: 'fakeDiagram1', uuid: 'fakeUUID-Loaded', success: true }],
+  last_updated: '2024-04-29T14:00:00Z',
+};
+
+const notLoadedDiagramCacheMetadata: DiagramCacheMetadata = {
+  diagrams: [
+    { name: 'fakeDiagram2', uuid: 'fakeUUID-Not-Loaded', success: true },
+  ],
+  last_updated: '2024-04-29T14:00:00Z',
+};
+
+const errorDiagramCacheMetadata: DiagramCacheMetadata = {
+  diagrams: [{ name: 'fakeDiagram2', uuid: 'fakeUUID-Loaded', success: false }],
+  last_updated: '2024-04-29T14:00:00Z',
+};
+
+const combinedDiagramCacheMetadata: DiagramCacheMetadata = {
+  diagrams: [
+    { name: 'fakeDiagram1', uuid: 'fakeUUID-Loaded', success: true },
+    { name: 'fakeDiagram2', uuid: 'fakeUUID-Not-Loaded', success: true },
+    { name: 'fakeDiagram2', uuid: 'fakeUUID-Loaded', success: false },
+  ],
+  last_updated: '2024-04-29T14:00:00Z',
+};
+
+// prettier-ignore
+const diagrams: Diagrams =
+  {
+    "fakeUUID-Loaded": {
+      loading: false,
+      content: base64ModelDiagram,
+    },
+    "fakeUUID-Not-Loaded": {
+      loading: true,
+      content: base64ModelDiagram,
+    }
+  }
+
+const data = {
+  model: mockModel,
+  project: mockProject,
+};
+
+export const Loading: Story = {
+  args: { diagramMetadata: undefined, loaderArray: Array(3).fill(0) },
+  decorators: [
+    moduleMetadata({
+      providers: [
+        { provide: MAT_DIALOG_DATA, useValue: data },
+        { provide: MatDialogRef, useValue: {} },
+      ],
+    }),
+  ],
+};
+
+export const DiagramLoaded: Story = {
+  args: {
+    diagramMetadata: loadedDiagramCacheMetadata,
+    diagrams: diagrams,
+  },
+  decorators: [
+    moduleMetadata({
+      providers: [
+        { provide: MAT_DIALOG_DATA, useValue: data },
+        { provide: MatDialogRef, useValue: {} },
```

Comment:
If you use providers for all stories, you can also define those on a module level.